source/client-backpressure/client-backpressure.md

```diff
@@ -0,0 +1,340 @@
+is safely retryable regardless of the type of operation, its metadata, or any of its arguments.
+
+#### SystemOverloadedError label
+
+An error is considered overloaded if it includes the "SystemOverloadError" label. This error label indicates that the
+server is overloaded. If this error label is present, drivers will backoff before attempting a retry.
+
+#### Overload Errors
+
+An overload error is any command or network error that occurs due to a server overload. For example, when a request
+exceeds the ingress request rate limit:
+
+```js
+{
+  'ok': 0.0,
+  'errmsg': "Rate limiter 'ingressRequestRateLimiter' rate exceeded",
+  'code': 462,
+  'codeName': 'IngressRequestRateLimitExceeded',
+  'errorLabels': ['SystemOverloadedError', 'RetryableError'],
+}
+```
+
+When a new connection attempt exceeds the ingress connection rate limit, the server closes the TCP connection before TLS
+handshake is complete. Drivers will observe this as a network error (e.g. "connection reset by peer" or "connection
+closed").
+
+When a new connection attempt is queued by the server for so long that the driver-side timeout expires, drivers will
+observe this as a network timeout error.
+
+Note that there is no guarantee that all SystemOverloaded errors are retryable or that all RetryableErrors also have the
+SystemOverloaded error label.
+
+#### Goodput
+
+The throughput of positive, useful output. In the context of drivers, this refers to the number of non-error results
+that the driver processes per unit of time.
+
+See [goodput](https://en.wikipedia.org/wiki/Goodput).
+
+### Requirements for Client Backpressure
+
+#### Overload retry policy
+
+This specification expands the driver's retry ability to all commands if the error indicates that it is both an overload
+error and that it is retryable, including those not currently considered retryable such as updateMany, create
+collection, getMore, and generic runCommand. The new command execution method obeys the following rules:
+
+1. If the command succeeds on the first attempt, drivers MUST deposit `RETRY_TOKEN_RETURN_RATE` tokens.
+    - The value is 0.1 and non-configurable.
+2. If the command succeeds on a retry attempt, drivers MUST deposit `RETRY_TOKEN_RETURN_RATE`+1 tokens.
+3. If a retry attempt fails with an error that does not include `SystemOverloadedError` label, drivers MUST deposit 1
+    token.
+    - A non-SystemOverloaded error indicates that the server is healthy enough to handle requests. For the purposes of
+        retry budget tracking, this counts as a success.
+4. A retry attempt will only be permitted if:
+    1. The error has both the `SystemOverloadedError` and the `RetryableError` label.
+    2. We have not reached `MAX_ATTEMPTS`.
+    3. (CSOT-only): `timeoutMS` has not expired.
+    4. (`SystemOverloadedError` errors only) a token can be acquired from the token bucket.
```

Comment:
@stIncMale TODO

Item 4.1 requires the `SystemOverloadedError` label to be present. The highlighted item 4.4 restricts itself to `SystemOverloadedError` errors only, but no other errors can possibly be considered because of item 4.1. Given this, why does item 4.4 has this restriction? Is this because we expect to change requirement 4.1 in the future, and are afraid we will forget to update requirement 4.4 with the restriction, which will become necessary at that point?